lib/concurrent/synchronization/java_pure_object.rb

```diff
@@ -32,6 +33,10 @@ def ns_signal
           JRuby.reference0(self).notify
           self
         end
+
+        def ensure_ivar_visibility!
+          # relying on undocumented behavior of JRuby, ivar access is volatile
```

Comment:
No changes at present. Most likely change we'd land after final would be the reified variables, but they'd have to get volatility treatment before release anyway.